### PR TITLE
Introduce ability to limit host-tags a project may use

### PR DESCRIPTION
**Pull request by doanac**

### CODE REVIEW

**Comment by doanac** [comment]
ci-retest

**Comment by doanac** [comment]
ci-retest

**Comment by doanac** [comment]
@vkhoroz - i know you lack some context for this. The general idea is to start creating projects where we explicitly configure what types of CI workers it has access. When creating *new* factories, I can set an explicit list.

File: jobserv/api/project.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Copyright (C) 2017 Linaro Limited
 # Author: Andy Doan <andy.doan@linaro.org>
 
+import json
```
Comment (vkhoroz) [inline jobserv/api/project.py:4]: Typo in commit message: availble -> available.

File: jobserv/trigger.py [hunk 2]
```diff
@@ -70,6 +76,14 @@ def trigger_runs(
         raise ApiError(500, str(e) + "\n" + traceback.format_exc())
 
 
+def _fail_run(run, reason):
+    run.set_status(BuildStatus.FAILED)
+    storage = Storage()
+    with storage.console_logfd(run, "a") as f:
+        f.write(reason)
+    storage.copy_log(run)
```
Comment (vkhoroz) [inline jobserv/trigger.py:84]: I guess this weird code is copied from elsewhere, right?
Do you want to reuse it instead?

File: jobserv/api/project.py [hunk 3]
```diff
@@ -50,7 +50,7 @@ def project_update(proj):
     p = get_or_404(Project.query.filter_by(name=proj))
 
     d = request.get_json() or {}
-    if "allowed-hosts-tags" in d:
+    if "allowed-host-tags" in d:
```
Comment (vkhoroz) [inline jobserv/api/project.py:53]: This looks like a typo fix of a previous commit. Squash?

File: jobserv/app.py [hunk 4]
```diff
@@ -337,6 +337,8 @@ def run_status(project, build, run, status=None):
                 db.session.delete(t)
         run.set_status(status)
         db.session.commit()
+        if run.status in (BuildStatus.FAILED, BuildStatus.PASSED):
+            Storage().copy_log(run)
```
Comment (vkhoroz) [inline jobserv/app.py:341]: Not sure I understand what's up here I mean: WHY?
Maybe, a one-line explaination in the commit message would be ok?

**Review by vkhoroz — COMMENTED** [review]
LGTM

File: jobserv/api/project.py (hunk 1 above)
Comment (doanac) [reply jobserv/api/project.py:4]: fixed in force push

File: jobserv/api/project.py (hunk 3 above)
Comment (doanac) [reply jobserv/api/project.py:53]: fixed in force push

File: jobserv/trigger.py (hunk 2 above)
Comment (doanac) [reply jobserv/trigger.py:84]: its kind of a pain to refactor it right now. i'll wait till it gets a little worse

File: jobserv/app.py (hunk 4 above)
Comment (doanac) [reply jobserv/app.py:342]: added a comment in the force push

**Review by vkhoroz — APPROVED** [review]
LGTM

**Pull request merged**